ctapipe/image/pixel_likelihood.py

```diff
@@ -177,13 +177,15 @@ def neg_log_likelihood(
     approx_mask = prediction > prediction_safety
 
     neg_log_l = 0
-    neg_log_l += neg_log_likelihood_approx(
-        image[approx_mask], prediction[approx_mask], spe_width, pedestal,
-    )
+    if np.sum(approx_mask) > 0:
```

Comment:
Use `np.count_nonzero` to check how many items are True in a mask. If you just need a single one, use `np.any`